dart_sys/src/lib.rs

```diff
@@ -0,0 +1,4395 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(unused_variables)]
+#![allow(dead_code)]
+
+mod tests;
+
+/* automatically generated by rust-bindgen */
```

Comment:
how is this file generated? IMHO we should:

1. include the file generation script as a part of source
2. in `ci.yaml`, run file generation script automagically, to ensure lib.rs does not change. To do this, just mimic current CI, where we run flutter_rust_bridge codegen automatically to ensure generated codes in frb_examples are up to date

Reply:
I truly have no idea how it was generated... Take a look at the crate's `Cargo.toml`.
https://github.com/OptimisticPeach/dart-sys/blob/master/Cargo.toml#L22

`bindgen` was included as a dev-dependency and at some point commented out. 

I think that it was probably originally generated but is now hand maintained...?

There is a directory at the root named `bindgen/` with a single file, `wrapper.h`.

Wrapper.h:
```c
#include "dart_api.h"
#include "dart_native_api.h"
#include "dart_tools_api.h"
```

There is nothing in the `build.rs` related to code generation, so my best guess is that these are all remnants from code that at some point was generated, but is now just maintained by hand.

Thoughts?

Reply:
> my best guess is that these are all remnants from code that at some point was generated, but is now just maintained by hand.

I also think so. However, to be a part of flutter_rust_bridge, we should ensure the quality of code. Thus, maybe try to firstly write down a simple script to invoke bindgen, see how the output differs, try to modify bindgen config, see again output, etc. if we have to modify sth by hand, would you please describe it here so we may find out a solution to avoid modifying by hand.

Reply:
please have a thorough look at the diff in this commit:
https://github.com/OptimisticPeach/dart-sys/commit/ecdb7ffc4fa3515dee163172e0e703c28914d964

this should answer questions on where the apparently generated parts of files came from, as well as why they are no longer needed to be generated.

Reply:
this means we can safely remove `bindgen/wrapper.h`

Reply:
Good spot :)

> why they are no longer needed to be generated.

Well IMHO we should still keep the generation script (which you have already found), and ensure the generated content is up-to-date via CI. This is analogy to: Say someone have a .rs file and compile it to .so, then we just throw away the .rs (given that it rarely changes) and only keep .so in the codebase - surely we should not do that ;)

Reply:
If it is hard to trigger the generation script, maybe wait for https://github.com/fzyzcjy/flutter_rust_bridge/pull/882 - which adds another crate to solely run bindgen

Reply:
Agreed 🤝